cmd/sidecars/qemu-args/main.go

```diff
@@ -0,0 +1,155 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package main
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/spf13/pflag"
+
+	vmSchema "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+const (
+	qemuCommandLineArgsAnnotation = "libvirt.vm.kubevirt.io/qemuArgs"
+	NS_SCHEMA_QEMU_1_0            = "http://libvirt.org/schemas/domain/qemu/1.0"
+)
+
+var logger *log.Logger
+
+func mergeQEMUCmd(current, annotations *api.Commandline) api.Commandline {
+	// Use map for simplicity
+	mapEnv := make(map[string]string)
+	mapArg := make(map[string]int)
+
+	// Populate map with current
+	for _, env := range current.QEMUEnv {
+		mapEnv[env.Name] = env.Value
+	}
+	for index, arg := range current.QEMUArg {
+		mapArg[arg.Value] = index
+	}
+
+	// Here we might overwrite existing enviroment values.
+	for _, env := range annotations.QEMUEnv {
+		if value, exists := mapEnv[env.Name]; exists && value != env.Value {
+			logger.Printf("Overwriting enviroment variable '%s': From '%s' to '%s'",
+				env.Name, value, env.Value)
+		}
+		mapEnv[env.Name] = env.Value
+	}
+
+	// Here we might warn if arg already exists.
+	for _, arg := range current.QEMUArg {
+		if _, exists := mapEnv[arg.Value]; exists {
```

Comment:
Probably this should check in `mapArg`:

```suggestion
		if _, exists := mapArg[arg.Value]; exists {
```